mailet/standard/src/main/java/org/apache/james/transport/matchers/SMTPAuthUserIs.java

```diff
@@ -57,11 +58,10 @@ public void init() throws javax.mail.MessagingException {
 
     @Override
     public Collection<MailAddress> match(Mail mail) {
-        String authUser = (String) mail.getAttribute(Mail.SMTP_AUTH_USER_ATTRIBUTE_NAME);
-        if (authUser != null && users.contains(authUser)) {
-            return mail.getRecipients();
-        } else {
-            return null;
-        }
+        return AttributeUtils
+            .getValueAndCastFromMail(mail, Mail.SMTP_AUTH_USER, String.class)
+            .filter(users::contains)
+            .map(ignored -> mail.getRecipients())
+            .orElse(null);
```

Comment:
it is safer to return an empty list if it doesn't break the logic.

Reply:
The API have to be rework, I want to keep the current behavior